tools/mo/openvino/tools/mo/moc_frontend/pipeline.py

```diff
@@ -53,10 +53,20 @@ def check_places_are_same(places_original: List[Place], places_new: List[Place])
         new_output_places = [x['node'] for x in outputs]
         log.debug('Using extract subgraph')
         input_model.extract_subgraph(new_input_places, new_output_places)
+        # invalidation of existing Place objects could have happened in the operation above
```

Comment:
good finding! Can there be some test added for this? Like that user needs to cut and reshape a model

Reply:
This is not tested by individual tests (yet) but a combination of MO params is used in many E2E validation pipelines already. I'll try adding some more synthetic E2E tests to another repo though.